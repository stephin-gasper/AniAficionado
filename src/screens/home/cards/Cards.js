import React from 'react';
import PropTypes from 'prop-types';
import {
  FeaturedImage,
  FeaturedImageWrapper,
  Title,
  Wrapper,
} from './Cards.style';

const Cards = ({imageUrl, title}) => (
  <Wrapper>
    <FeaturedImageWrapper>
      <FeaturedImage
        source={{
          uri: imageUrl,
        }}
      />
    </FeaturedImageWrapper>
    <Title>{title}</Title>
  </Wrapper>
);

Cards.propTypes = {
  imageUrl: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default Cards;
