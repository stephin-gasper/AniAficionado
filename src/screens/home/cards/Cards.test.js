import React from 'react';
import {render} from '@testing-library/react-native';

import Cards from './Cards';

describe('<Cards />', () => {
  const defaultProps = {
    imageUrl: 'https://www.sample-img.com',
    title: 'Sample title',
  };

  it('should render correctly', () => {
    expect(render(<Cards {...defaultProps} />)).toMatchSnapshot();
  });
});
