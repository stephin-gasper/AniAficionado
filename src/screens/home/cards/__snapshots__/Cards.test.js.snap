// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cards /> should render correctly 1`] = `
<View
  style={
    Array [
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Image
      source={
        Object {
          "uri": "https://www.sample-img.com",
        }
      }
      style={
        Array [
          Object {},
        ]
      }
    />
  </View>
  <Text
    style={
      Array [
        Object {},
      ]
    }
  >
    Sample title
  </Text>
</View>
`;
